presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushDownTableConstraints.java

```diff
@@ -136,15 +142,17 @@ public PushDownTableConstraints(Metadata metadata)
         };
     }
 
-    private boolean planChanged(PlanNode rewrittenPlan, FilterNode oldPlan, Lookup lookup)
+    private boolean planChanged(PlanNode rewrittenPlan, FilterNode oldPlan, Lookup lookup, Session session, SymbolAllocator symbolAllocator)
     {
         if (!(rewrittenPlan instanceof FilterNode)) {
             return true;
         }
 
         FilterNode rewrittenFilter = (FilterNode) rewrittenPlan;
-        if (!rewrittenFilter.getPredicate().equals(oldPlan.getPredicate())) {
-            return true;
+        if (!new ExpressionEquivalence(metadata, sqlParser).areExpressionsEquivalent(session, rewrittenFilter.getPredicate(), oldPlan.getPredicate(), symbolAllocator.getTypes())) {
```

Comment:
Isn't there a way to check the equivalence without using Parser? Is it because we still use the same nodes/classes from parser/AST?